### PR TITLE
Make `pyproject.toml` comments inline

### PR DESCRIPTION
**Pull request by user27182**
### Overview

Update comments in `pyproject.toml` in preparation for #6781.  This PR ensures that comments will stay relevant and are correctly moved along with any sorted values.

### Details
- Move comments which apply to a single line to be inline
- Duplicate comments which apply to multiple unsorted lines and make them inline. With #6781, the arrays will be sorted, and any comments will also be moved as part of the sorting. This creates issues if a single comment was previously used to refer to multiple unsorted lines, since the comment will only follow the first entry.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -300,53 +297,34 @@ quote-style = "single"
...
     "D107", # Missing docstring in `__init__`
     "D203", # May conflict with the formatter
     "D211", # Incompatible with D203
     "D213", # Incompatible with D212
     "D402", # First line should not be the function's signature
     "D416", # Section name ends in colon
-    # whitespace before ':'
-    "E203",
-    # line break before binary operator
-    # "W503",
```
Comment (user27182) [inline pyproject.toml:317]: Note: This comment is removed completely
```
# line break before binary operator
# "W503",
```

File: pyproject.toml [hunk 2]
```diff
@@ -300,53 +297,34 @@ quote-style = "single"
...
-    # too many leading '#' for block comment
-    "E266",
-    # ambiguous variable name
-    "E741",
-    # module level import not at top of file
-    "E402",
-    # Quotes (temporary)
-    "Q0",
-    # bare excepts (temporary)
-    # "B001", "E722",
```
Comment (user27182) [inline pyproject.toml:331]: Note: This comment is removed completely
```
# bare excepts (temporary)
# "B001", "E722",
```

File: pyproject.toml [hunk 3]
```diff
@@ -273,9 +272,7 @@ checks = [
 
 exclude = [  # don't report on objects that match any of these regex
     '\._.*$',  # Ignore anything that's private (e.g., starts with _)
-    # classes inherit from BaseReader
-    '\.*Reader(\.|$)',
-    # Documentation extensions or utilities
```
Comment (user27182) [inline pyproject.toml:278]: Note: This comment is removed completely.
```# Documentation extensions or utilities```

**Review by tkoyama010 — APPROVED** [review]
LGTM

**Pull request merged**